packages/cadl-ranch-specs/http/parameters/spread/main.tsp

```diff
@@ -224,40 +260,85 @@ namespace Alias {
     @header
     `x-ms-test-header`: string;
 
-    prop1: string;
-    prop2: string;
-    prop3: string;
-    prop4: string;
-    prop5: string;
-    prop6: string;
+    /** required string */
+    requiredString: string;
+
+    /** optional int */
+    optionalInt?: int32;
+
+    /** required int */
+    requiredIntList: int32[];
+
+    /** optional string */
+    optionalStringList?: string[];
   };
 
   @scenario
   @scenarioDoc("""
   Test case for spread alias including 6 parameters. May handle as property bag for these parameters.
   
-  Should not generate any model named `AliasMultipleRequestParameters`.
-  Should generate an operation like below:
-  ```
-  spreadWithMultipleParameters(id: string, x_ms_test_header: string, prop1: string, prop2: string, prop3: string, prop4: string, prop5: string, prop6: string)
-  ```
+  Should not generate any model named `MultipleRequestParameters`.
+
+  Since it contains both optional properties and required properties, the method signature might vary across different languages.
+
   Note it's also acceptable if some languages handle it as property bag.
   
   Expected path parameter: id="1"
   Expected header parameter: x-ms-test-header="bar"
   Expected request body:
   ```json
   {
-    "prop1":"foo1",
-    "prop2":"foo2",
-    "prop3":"foo3",
-    "prop4":"foo4",
-    "prop5":"foo5",
-    "prop6":"foo6"
+    "requiredString": "foo",
+    "optionalInt": 1,
+    "requiredIntList": [1, 2],
+    "optionalStringList": ["foo", "bar"]
+
   }
   ```
   """)
   @route("/multiple-parameters/{id}")
   @put
   op spreadWithMultipleParameters(...MultipleRequestParameters): NoContentResponse;
+
+  alias InnerAlias = {
+    @doc("name of the Thing")
+    name: string;
+
+    @doc("age of the Thing")
+    age: int32;
+  };
+
+  alias InnerAliasParameter = {
+    @path id: string;
+    ...InnerAlias;
+
+    @header
+    `x-ms-test-header`: string;
+  };
+
+  @scenario
+  @scenarioDoc("""
+  Test case for spread alias with contains another alias property as body.
+
+  Should not generate any model named `InnerAlias` and `InnerAliasParameter`.
+  Should generate an operation like below:
+  ```
+  spreadAliasWithSpreadAlias(id: string, name: string, age: int32, top: int32)
+  ```
+  Note the parameter name is guessed from the model name and it may vary by language.
+  Expected path parameter: id="1"
+  Expected header parameter: x-ms-test-header="bar"
+  Expected request body:
+  ```json
+  {
+    "name": "foo",
+    "age": 1
+  }
+  ```
+  """)
+  @route("/request-with-spread-alias")
+  @doc("spread an alias with contains another alias property as body.")
+  @post
+  @convenientAPI(true)
+  op spreadAliasWithSpreadAlias(...InnerAliasParameter): void;
```

Comment:
Why not align with others?
 ```suggestion
  op spreadAsInnerAliasParameter(...InnerAliasParameter): void;
```
Same as the route.

Reply:
The error in this part was caused by different work progress across my devices. I have already used VSCode to compare the differences between the new and old code and have synchronized them.